ffs/api/api.go

```diff
@@ -96,9 +106,43 @@ func (i *API) ID() ffs.APIID {
 	return i.cfg.ID
 }
 
-// WalletAddr returns the Lotus wallet address.
-func (i *API) WalletAddr() string {
-	return i.cfg.WalletAddr
+// WalletAddr returns the wallet addresses.
+func (i *API) WalletAddrs() map[string]AddrInfo {
+	return i.cfg.Addrs
+}
+
+func (i *API) GetDefaultCfg() ffs.DefaultCidConfig {
+	return i.cfg.DefaultCidConfig
+}
+
+// NewAddr creates a new address managed by the FFS instance
+func (i *API) NewAddr(name string, options ...NewAddressOption) (string, error) {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	conf := &NewAddressConfig{
+		makeDefault: false,
+		addressType: defaultAddressType,
+	}
+	for _, option := range options {
+		option(conf)
+	}
+
+	addr, err := i.wm.NewAddress(context.Background(), conf.addressType)
+	if err != nil {
+		return "", fmt.Errorf("creating new wallet addr: %s", err)
+	}
+	i.cfg.Addrs[addr] = AddrInfo{
```

Comment:
I think we should check if that `name` wasn't used before and fail if that's the case. To avoid an accidental override.

Reply:
We can do that. Just to be clear, a duplicate name won't override any other address since they are keyed by address, not name. But you would end up with duplicate names which may not be ideal.

Reply:
Agree!